src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -756,12 +800,38 @@ private static bool IsTypeLessVisibleThan(TypeSymbol type, Symbol sym, ref Compo
             bool canDigThroughNullable = false,
             bool useDefaultType = false,
             bool visitCustomModifiers = false)
+        {
+            var matchigSymbols = VisitType(
+                typeWithAnnotationsOpt: typeWithAnnotationsOpt,
+                type: type,
+                typeWithAnnotationsPredicate: typeWithAnnotationsPredicate,
+                typePredicate: typePredicate,
+                arg: arg,
+                returnOnMatch: true,
+                canDigThroughNullable: canDigThroughNullable,
+                useDefaultType: useDefaultType,
+                visitCustomModifiers: visitCustomModifiers);
+            return matchigSymbols.FirstOrDefault();
+        }
+
+        public static List<TypeSymbol> VisitType<T>(
+            this TypeWithAnnotations typeWithAnnotationsOpt,
+            TypeSymbol? type,
+            Func<TypeWithAnnotations, T, bool, bool>? typeWithAnnotationsPredicate,
+            Func<TypeSymbol, T, bool, bool>? typePredicate,
+            T arg,
+            bool returnOnMatch,
+            bool canDigThroughNullable = false,
+            bool useDefaultType = false,
+            bool visitCustomModifiers = false)
         {
             RoslynDebug.Assert(typeWithAnnotationsOpt.HasType == (type is null));
             RoslynDebug.Assert(canDigThroughNullable == false || useDefaultType == false, "digging through nullable will cause early resolution of nullable types");
             RoslynDebug.Assert(canDigThroughNullable == false || visitCustomModifiers == false);
             RoslynDebug.Assert(visitCustomModifiers == false || typePredicate is { });
 
+            List<TypeSymbol> combinedResult = [];
```

Comment:
It feels like this could hurt perf if this method (`VisitType`) is used often which I think it is.